meta-oe/recipes-devtools/perfetto/perfetto.bb

```diff
@@ -0,0 +1,145 @@
+LICENSE = "Apache-2.0 & Unknown"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=f87516e0b698007e9e75a1fe1012b390 \
+                    file://buildtools/android-unwinding/libunwindstack/LICENSE_BSD;md5=6e0dc88b22ba6908afbc1a1d8abbb148 \
+                    file://buildtools/benchmark/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
+                    file://buildtools/googletest/LICENSE;md5=cbbd27594afd089daa160d3a16dd515a \
+                    file://buildtools/jsoncpp/LICENSE;md5=fa2a23dd1dc6c139f35105379d76df2b \
+                    file://buildtools/jsoncpp/devtools/licenseupdater.py;md5=891928af1911fc7d5a4383b63dd0ae66 \
+                    file://buildtools/libbacktrace/LICENSE;md5=24b5b3feec63c4be0975e1fea5100440 \
+                    file://buildtools/libcxx/LICENSE.TXT;md5=55d89dd7eec8d3b4204b680e27da3953 \
+                    file://buildtools/libcxx/utils/google-benchmark/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
+                    file://buildtools/libcxxabi/LICENSE.TXT;md5=7b9334635b542c56868400a46b272b1e \
+                    file://buildtools/libunwind/LICENSE.TXT;md5=f66970035d12f196030658b11725e1a1 \
+                    file://buildtools/linenoise/LICENSE;md5=faa55ac8cbebebcb4a84fe1ea2879578 \
+                    file://buildtools/lzma/Java/Tukaani/COPYING;md5=3d9c8c11183717c7a0539a9afccec6e4 \
+                    file://buildtools/protobuf/LICENSE;md5=37b5762e07f0af8c74ce80a8bda4266b \
+                    file://buildtools/sqlite_src/LICENSE.md;md5=82c5c312251c7bb3f775273ef25592e2 \
+                    file://buildtools/sqlite_src/autoconf/tea/license.terms;md5=5e0405ae78edb434653628790bb02b17 \
+                    file://buildtools/zlib/LICENSE;md5=f09575dbfb09420642318b413159496f \
+                    file://debian/copyright;md5=4e08364c82141f181de69d0a2b89d612 \
+                    file://python/LICENSE;md5=c602a632c34ade9c78a976734077bce7"
+
+SRC_URI = "git://github.com/google/perfetto.git;protocol=https"
+SRC_URI[sha256sum] = "3d92dc8f561ea6a83f3efdea33c5aeefa4a3cd98f9739190efd9c33c30bcbdf0"
+SRCREV = "5bd3f582c075d0d026c5fe0b5e291d34dee0d976"
+
+DEPENDS += " ninja-native"
+DEPENDS += " curl-native"
+DEPENDS += " ca-certificates-native"
+
+# To avoid the GNU_HASH warning
+TARGET_CC_ARCH += "${LDFLAGS}"
+
+S = "${WORKDIR}/git"
+
+FILES:${PN}:append = " \
+  ${bindir}/tracebox \
+  "
+
+# Perfetto includes an "install-build-deps" script. It requires network connection.
```

Comment:
this should be perhaps improved. We need to express the SRC_URIs to include all thats needed. The current do_fetch seems a bit clunky

Reply:
Could you explain does `express the SRC_URIs to include all thats needed` mean? I can also look into other recipes to see how this is handled.

It is now clunky because I need to run `python3 tools/install-build-deps` in the fetch step (which requires network connection, so it has to be in the fetch step), and have Bitbake pack the source tree plus the installed dependencies into an archive.

Reply:
But calling this script in do_fetch isn't the same as bitbake fetcher fetching it from SRC_URI. Because bitbake fetcher will respect MIRROR/PREMIRROR variables and store the archives in DL_DIR, so next time you build this recipe it won't need to reach the network and just unpack from local DL_DIR.

You need to replicate what this script does (copy the URLs and revision etc) and then persuade the build that the sources are already downloaded (sometimes it's easy and the script just checks that the deps are there, sometimes there are extra variables you can set and sometimes you have to patch the build to allow this behavior. I haven't looked at what perfetto does exactly, but I had to deal with similar issues in meta-ros quite often, you can find some examples there.

Reply:
Some small updates: I have made some progress after realizing `SRC_URI` can contain multiple sources/repositories. I'm now trying to figure out how to properly do the "unpack" step. The default unpack seems to overwrite whatever has already been "unpacked" so I'm looking for a way to achieve the desired unpack behavior (copy all dependencies like googletest, protobuf, libcxxabi, etc. into Perfetto.)

Reply:
So I asked around on the OpenBMC Discord channel and managed to set SRC_URI and eliminated the custom fetch and unpack steps.